dashboard/src/components/BuildsTable/BuildsTable.tsx

```diff
@@ -234,33 +291,28 @@ export function BuildsTable({
         </TableRow>
       );
     }
-  }, [
-    columns.length,
-    groupHeaders.length,
-    modelRows,
-    onClickShowBuild,
-    openLogSheet,
-  ]);
+  }, [columns.length, modelRows, TableRowMemoized]);
 
   const handlePreviousItem = useCallback(() => {
-    setLog(c => {
-      if (c !== null && c > 0) return c - 1;
+    setLog(previousLog => {
+      if (previousLog !== null && previousLog > 0) return previousLog - 1;
 
-      return c;
+      return previousLog;
     });
   }, [setLog]);
 
   const handleNextItem = useCallback(() => {
-    setLog(c => {
-      if (c !== null && c < sortedItems.length - 1) return c + 1;
+    setLog(previousLog => {
+      if (previousLog !== null && previousLog < sortedItems.length - 1)
+        return previousLog + 1;
 
-      return c;
+      return previousLog;
     });
   }, [setLog, sortedItems.length]);
 
   const { data: dataBuildCount, isLoading } = useBuildStatusCount(
-    { buildId: sortedItems[currentLog ?? 0].id ?? '' },
-    { enabled: !!sortedItems[currentLog ?? 0].id },
+    { buildId: sortedItems.length > 0 ? sortedItems[currentLog ?? 0].id : '' },
+    { enabled: sortedItems.length > 0 && !!sortedItems[currentLog ?? 0].id },
```

Comment:
```suggestion
    { buildId: sortedItems.length > 0 ? sortedItems[currentLog ?? 0]?.id : '' },
    { enabled: sortedItems.length > 0 && !!sortedItems[currentLog ?? 0]?.id },
```
array access can always be undefined, we need to check them